File: lessons/compiled_lang/cs/src/h00.associative.cs
using System;
using System.Collections.Generic;

class H00Associative
{
    static void Main()
    {
        // create empty dictionary
        var ages = new Dictionary<string, int>();
        // insert one element at a time
        ages["bob"] = 34;
        ages["ed"] = 58;
        ages["steve"] = 32;
        ages["ralph"] = 23;
        ages["deb"] = 46;
        ages["kate"] = 19;

        // enumerate and print keys
        Console.WriteLine("Keys (names):  " + string.Join(", ", ages.Keys));
        // enumerate and print values
        Console.WriteLine("Values (ages): " + string.Join(", ", ages.Values));
    }
}

File: lessons/compiled_lang/rust/src/h00.associative.rs
use std::collections::HashMap;

fn main() {
    // create empty map
    let mut ages: HashMap<&str, i32> = HashMap::new();
    // insert one element at a time
    ages.insert("bob", 34);
    ages.insert("ed", 58);
    ages.insert("steve", 32);
    ages.insert("ralph", 23);
    ages.insert("deb", 46);
    ages.insert("kate", 19);

    // enumerate and print keys and values
    let keys: Vec<String> = ages.keys().map(|k| k.to_string()).collect();
    let values: Vec<String> = ages.values().map(|v| v.to_string()).collect();
    println!("Keys (names):  {}", keys.join(", "));
    println!("Values (ages): {}", values.join(", "));
}
